File: Python/rulemsx-1.0.1/rulemsx/action.py
'''
Created on 27 Nov 2017

@author: RCLEGG2@BLOOMBERG.NET
'''

import logging

class Action:
    
    def __init__(self, name, executor=None):
        
        logging.info("Initializing Action: " + name)
        
        self.name = name
        self.action_executor = executor

        logging.info("Initialized Action: " + name)
          
    def execute(self, dataset):
        raise NotImplementedError("The execute function of an Action must be overridden")

        
__copyright__ = """
Copyright 2017. Bloomberg Finance L.P.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to
deal in the Software without restriction, including without limitation the
rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
sell copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:  The above
copyright notice and this permission notice shall be included in all copies
or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
IN THE SOFTWARE.
"""

File: CSharp/cs_RuleMSX-master/RuleMSX/Action.cs
namespace com.bloomberg.samples.rulemsx
{
    public class Action
    {
        string name;
        ActionExecutor executor;

        internal Action(string name)
        {
            this.name = name;
        }
        internal Action(string name, ActionExecutor executor)
        {
            this.name = name;
            this.addExecutor(executor);
        }

        public void addExecutor(ActionExecutor executor)
        {
            this.executor = executor;
        }

        public string getName()
        {
            return this.name;
        }

        public ActionExecutor getExecutor()
        {
            return this.executor;
        }
    }
}
